components/org.wso2.carbon.identity.api.user.governance/src/main/java/org/wso2/carbon/identity/user/endpoint/util/Utils.java

```diff
@@ -542,6 +550,68 @@ private static void logException(ErrorDTO error, Throwable cause, Log log) {
         log.error(errorMsg, cause);
     }
 
+    /**
+     * Mask the given value if it is required.
+     *
+     * @param value Value to be masked.
+     * @return Masked/unmasked value.
+     */
+    public static String maskIfRequired(String value) {
+
+        return LoggerUtils.isLogMaskingEnable ? LoggerUtils.getMaskedContent(value) : value;
+    }
+
+    /**
+     * Retrieves the claim value for a user from the user store.
+     *
+     * @param user     User object containing user information
+     * @param claimURI The URI of the claim to retrieve
+     * @param log      Log instance for logging errors
+     * @return The claim value, or an empty string if the claim is not found or an error occurs
+     */
+    public static String getUserClaim(UserDTO user, String claimURI, Log log) {
```

Comment:
What would be the better way on passing loggers. I feel we can user the logger object in this util class rather than passing a logger object. Is there such referenced?

Reply:
We can make this util class reusable by improving the method to expect username, claimURI, tenantDomain

You can explain that username should be domain qualified in the method comments.

When calling this method, the domain qualified username has to be passed. So this method can be reusable in the future.

Reply:
Resolved via https://github.com/wso2-extensions/identity-governance/pull/936/commits/b9aaa32e67b61ec7fba66cc9c9449a16723639fb